src/Simplify.cpp

```diff
@@ -4983,6 +5017,16 @@ void check_boolean() {
     check(broadcast(b1, 4) && broadcast(b1, 4), broadcast(b1, 4));
     check(broadcast(b1, 4) || broadcast(b1, 4), broadcast(b1, 4));
 
+    check((x == 1) && (x != 2), (x == 1) && (1 != 2));
```

Comment:
The right-hand-sides here should be the fully-simplified forms, so shouldn't the RHS here be just (x == 1) (and similar below)?